src/python/pants/backend/python/util_rules/interpreter_constraints_test.py

```diff
@@ -263,3 +266,47 @@ def test_group_field_sets_by_constraints_with_unsorted_inputs() -> None:
             Address("src/python/c_dir/path.py", target_name="test"), "==3.6.*"
         ),
     )
+
+
+@pytest.mark.parametrize(
+    "nums,expected",
+    (
+        ([0, 1, 2], []),
+        ([0, 2], [1]),
+        ([0, 1, 2, 4], [3]),
+        ([0, 1, 4], [2, 3]),
+        ([1, 4], [2, 3]),
+        ([1, 2, 3, 4], []),
+    ),
+)
+def test_not_in_contiguous_range(nums, expected) -> None:
+    assert _not_in_contiguous_range(nums) == expected
+
+
+@pytest.mark.parametrize(
+    "constraints,expected",
+    (
+        # Input constraints already are for a single interpreter version.
+        (["==3.7.5"], [["==3.7.5"]]),
+        (["==3.7.*"], [[">=3.7,<3.8"]]),
```

Comment:
Sure, I can fix to use a wildcard if the whole interpreter is supported. I was using `">=3.7,<3.8"` because it's simpler code to create, but I'll bias towards a clearer UX than simpler code.